src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -646,10 +646,4 @@ private String getJenkinsVersion(){
         }
         return this.jenkinsVersion;
     }
-
-    private static boolean isCollectBuildLogEnabled(){
-        return DatadogUtilities.getDatadogGlobalDescriptor() != null &&
```

Comment:
So we're not checking for null anymore which is the reason why you had to add https://github.com/jenkinsci/datadog-plugin/pull/85/files#diff-16902ca2fc3a4b1fa3f87d6ad966d842R43 in the tests.
I'm not totally confident in the statement "It can only throw a IllegalStateException when running tests" [here](https://github.com/jenkinsci/datadog-plugin/pull/85/files#diff-6247ca3477c9a4d9358a5c115489c389R60). If the DatadogGlobalConfiguration extension fails to load for any reason, the singleton will be null.

Reply:
We aren't checking for null as part of the runtime code, correct.  It shouldn't be necessary to guard against that since protecting from a failed config load will just hide other problems.

`DatadogUtilities` do a lot of checking for null which we should actually remove too, but there is deeper logic in our tests that currently depend on the default responses which makes it a more involved change.

The general approaches seem to have this behavior as well - see for instance this line: https://github.com/jenkinsci/azure-keyvault-plugin/blob/ef3a5eb4413f7ecd47d6532735604c52fceaaef0/src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultBuildWrapper.java#L136